### PR TITLE
Closes #409: Read and merge two DocumentToProject sources for affmatching

### LINKED ISSUE #409: Join two document-project datastores before sending them to affiliations matching
One of them contains relations coming directly from InfoSpace with the following schema:

https://github.com/openaire/iis/blob/cdh5/iis-schemas/src/main/avro/eu/dnetlib/iis/importer/DocumentToProject.avdl

the other one is produced by `project-referenceextraction` algorithm and comes with the different schema:

https://github.com/openaire/iis/blob/cdh5/iis-schemas/src/main/avro/eu/dnetlib/iis/referenceextraction/project/DocumentToProject.avdl

The outcome should contain `confidenceLevel` field which means it should share `project-referenceextraction` schema.

This could be implemented either as independent module or become part of affiliations reader model.

### PR DESCRIPTION
**Pull request by madryk**

### CODE REVIEW

File: iis-wf/iis-wf-affmatching/src/test/resources/data/input/affiliations.json [hunk 1]
```diff
@@ -87,14 +87,14 @@
...
-        "organization":"Department of Molecular and Cell Biology, University in California at Berkeley",
+        "organization":"Department of Molecular and Cell Biology, University of California at Berkeley",
         "countryName":"United States",
         "countryCode":"US",
         "address":null,
         "rawText":""
         },
         {
-        "organization":"Department of Demography, University of California at Berkeley",
+        "organization":"Department of Demography, Unversity of California at Berkeley",
```
Comment (marekhorst) [inline iis-wf/iis-wf-affmatching/src/test/resources/data/input/affiliations.json]: Was it renamed on purpose?

File: iis-wf/iis-wf-affmatching/src/test/resources/data/input/affiliations.json [hunk 2]
```diff
@@ -128,7 +128,34 @@
     "authors":null,
     "affiliations":[
         {
-        "organization":"Department of Molecular and Cell Biology, University in California at Berkeley",
+        "organization":"Department of Molecular and Cell Biology, Unversity of California at Berkeley",
```
Comment (marekhorst) [inline iis-wf/iis-wf-affmatching/src/test/resources/data/input/affiliations.json]: Was it renamed on purpose?

File: iis-wf/iis-wf-affmatching/src/test/resources/data/input/affiliations.json [hunk 3]
```diff
@@ -128,7 +128,34 @@
...
+    "keywords":null,
+    "externalIdentifiers":null,
+    "journal":null,
+    "year":null,
+    "publisher":null,
+    "references":null,
+    "authors":null,
+    "affiliations":[
+        {
+        "organization":"Department of Molecular and Cell Biology, Unversity of California at Berkeley",
```
Comment (marekhorst) [inline iis-wf/iis-wf-affmatching/src/test/resources/data/input/affiliations.json]: Was it renamed on purpose?

**Comment by madryk** [comment]
All occurrences of word `Unversity` is some kind of a trick to assure matching will be from first bucket.
Second and third bucket will not match this name to proper organization (because proper organization will be in different bucket).
Before to achieve that I used `in` instead of `of` after `University`, but I decided that this typo is easier to notice.
To summarize
In `DOC_5` we have typo and too low `confidenceLevel` in `docProjInferred.json` - no expected match.
In `DOC_4` we have typo and too low `confidenceLevel` in `docProjInferred.json`, but there is relation in `docProj.json` - there should be match
In `DOC_3` pos `1` we have typo and `confidenceLevel` is good enough - there should be match
`DOC_3` pos `0` - is here to test higher value of `matchStrength` (two same matched organization in single document)

File: iis-wf/iis-wf-affmatching/src/test/resources/data/input/affiliations.json (hunk 1 above)
Comment (madryk) [inline iis-wf/iis-wf-affmatching/src/test/resources/data/input/affiliations.json]: Explanation below.

File: iis-wf/iis-wf-affmatching/src/test/resources/data/input/affiliations.json (hunk 2 above)
Comment (madryk) [inline iis-wf/iis-wf-affmatching/src/test/resources/data/input/affiliations.json]: Explanation below.

File: iis-wf/iis-wf-affmatching/src/test/resources/data/input/affiliations.json (hunk 3 above)
Comment (madryk) [inline iis-wf/iis-wf-affmatching/src/test/resources/data/input/affiliations.json]: Explanation below.

**Pull request closed without merging**